lib/aws-s3.js

```diff
@@ -403,19 +403,28 @@ module.exports = class AWSAttachmentsService extends require("./basic") {
    * @param {import('@sap/cds').Request} req - The request object
    */
   async attachDeletionData(req) {
-    const attachments = cds.model.definitions[req?.target?.name + ".attachments"]
-    if (attachments) {
-      const diffData = await req.diff()
-      let deletedAttachments = []
-      diffData.attachments?.filter((object) => {
-        return object._op === "delete"
-      })
-        .map((attachment) => {
-          deletedAttachments.push(attachment.ID)
+    const attachmentCompositions = Object.keys(req?.target?.associations).filter(assoc => req?.target?.associations[assoc]._target['@_is_media_data']);
+    if (attachmentCompositions.length) {
+      const diffData = await req.diff();
+      const queries = [];
+      for (const attachmentsComp of attachmentCompositions) {
+        let deletedAttachments = []
+        diffData[attachmentsComp]?.forEach(object => {
+          if (object._op === "delete") {
+            deletedAttachments.push(object.ID)
+          }
         })
```

Comment:
since lambda expressions already used there shouldn't be much of performance impact if using:

```suggestion
        let deletedAttachments = diffData[attachmentsComp]?
          .filter(object => object._op === "delete")?
          .map(object => object.ID) ?? []
```

Reply:
```suggestion
        const deletedAttachments = diffData[attachmentsComp]?
          .map(object => object._op === "delete" ? object.ID : null)?
          .filter(object => object) ?? []
```